File: leetcode/349.两个数组的交集/345.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        if(nums1 == null || nums1.length ==0 || nums2 == null || nums2.length == 0)
            return new int[0];
        Set<Integer> result = new HashSet<>();
        Set<Integer> cnt = new HashSet<>();
        for(int i:nums1){
            cnt.add(i);
        }
        for(int i:nums2){
            if(cnt.contains(i)){
                result.add(i);
            }
        }
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: leetcode/349.两个数组的交集/345.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        tab = {}
        for n in nums1:
            tab[n] = tab.get(n, 0) + 1
        result = set()
        for n in nums2:
            if n in tab:
                result.add(n)
                del tab[n]
        return list(result)